src/multicluster/components/MulticlusterYAMLCreation/hooks/useModelFromParam.ts

```
@@ -0,0 +1,18 @@
+import { useMatch } from 'react-router-dom-v5-compat';
+
+import { K8sModel, useK8sModel } from '@openshift-console/dynamic-plugin-sdk';
+
+const useModelFromParam = (): [model: K8sModel | null, loading: boolean] => {
+  const namespacedMatch = useMatch('/k8s/cluster/:cluster/ns/:ns/:modelRef/*');
+  const clusterwideMatch = useMatch('/k8s/cluster/:cluster/:modelRef/*');
+
+  const modelRef = namespacedMatch?.params?.modelRef || clusterwideMatch?.params?.modelRef || '';
+
+  const [group, version, kind] = modelRef?.split('~');
```

Biome (2.1.2)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)